citesphere-importer/src/main/java/edu/asu/diging/citesphere/importer/core/service/impl/CrossrefReferenceImportProcessor.java

```diff
@@ -0,0 +1,163 @@
+        itemTypeMapping.put(CrossRefPublication.REFERNCE_ENTRY, ItemType.DICTIONARY_ENTRY);
+        itemTypeMapping.put(CrossRefPublication.POSTED_CONTENT, ItemType.WEBPAGE);
+        itemTypeMapping.put(CrossRefPublication.COMPONENT, ItemType.ATTACHMENT);
+        itemTypeMapping.put(CrossRefPublication.EDITED_BOOK, ItemType.BOOK);
+        itemTypeMapping.put(CrossRefPublication.PROCEEDINGS_ARTICLE, ItemType.CONFERENCE_PAPER);
+        itemTypeMapping.put(CrossRefPublication.DISSERTATION, ItemType.THESIS);
+        itemTypeMapping.put(CrossRefPublication.BOOK_SECTION, ItemType.BOOK_SECTION);
+        itemTypeMapping.put(CrossRefPublication.REPORT_COMPONENT, ItemType.REPORT);
+        itemTypeMapping.put(CrossRefPublication.REPORT, ItemType.REPORT);
+        itemTypeMapping.put(CrossRefPublication.PEER_REVIEW, ItemType.JOURNAL_ARTICLE);
+        itemTypeMapping.put(CrossRefPublication.BOOK_TRACK, ItemType.BOOK);
+        itemTypeMapping.put(CrossRefPublication.BOOK_PART, ItemType.BOOK_SECTION);
+        itemTypeMapping.put(CrossRefPublication.OTHER, ItemType.DOCUMENT);
+        itemTypeMapping.put(CrossRefPublication.JORUNAL_VOLUME, ItemType.JOURNAL_ARTICLE);
+        itemTypeMapping.put(CrossRefPublication.BOOK_SET, ItemType.BOOK);
+        itemTypeMapping.put(CrossRefPublication.JOURNAL, ItemType.JOURNAL_ARTICLE);
+        itemTypeMapping.put(CrossRefPublication.PROCEEDINGS_SERIES, ItemType.CONFERENCE_PAPER);
+        itemTypeMapping.put(CrossRefPublication.REPORT_SERIES, ItemType.REPORT);
+        itemTypeMapping.put(CrossRefPublication.PROCEEDINGS, ItemType.CONFERENCE_PAPER);
+        itemTypeMapping.put(CrossRefPublication.DATABASE, ItemType.DATABASE);
+        itemTypeMapping.put(CrossRefPublication.STANDARD, ItemType.STATUTE);
+        itemTypeMapping.put(CrossRefPublication.REFERENCE_BOOK, ItemType.DICTIONARY_ENTRY);
+        itemTypeMapping.put(CrossRefPublication.GRANT, ItemType.DOCUMENT);
+        itemTypeMapping.put(CrossRefPublication.DATASET, ItemType.DATABASE);
+        itemTypeMapping.put(CrossRefPublication.BOOK_SERIES, ItemType.BOOK);
+    }
+    
+    public void startImport(KafkaJobMessage message, JobInfo info) {
+        // message = jobToken: eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJKT0IxOTciLCJleHAiOjE2NzgzNDU3NDF9.5Xqh_AoMHcdlatULkCLFtny9pOF_uJ-SRARw0gCybY3h3qHL2mkIIQlk-qTA0Pn0VlhOLuW4FwACHmIdwZVmoA
+        // info = dois: [10.2307/j.ctvcm4h07.67, 10.1515/9780691242507]
+        // null
+        // zotero: byRZjIk2y4e3kay1cnwy3KpB
+        // zoteroId: 9154965
+        logger.info("Starting import for " + info.getDois());
+        
+        ObjectMapper mapper = new ObjectMapper();
+        ArrayNode root = mapper.createArrayNode();
+        int entryCounter = 0;
+
+        sendMessage(null, message.getId(), Status.PROCESSING, ResponseCode.P00);
+        BibEntryIterator bibIterator = null;
+        try {
+            bibIterator = handlerRegistry.handleFile(info, null);
```

Comment:
why would the "handleFile" method be called here? there is no file.

Reply:
Replaced the logic.